src/native_main.nim

```diff
@@ -165,26 +166,44 @@ proc handleMessage(msg: MessageRecv): MessageResp =
         of "move":
             let src = expandTilde(msg.`from`.get())
             let dst = expandTilde(msg.to.get())
+            let canMove = msg.overwrite.get(false) or not(fileExists(dst) or
+                    fileExists(joinPath(dst, extractFilename(src))))
 
-            if fileExists(dst) or fileExists(joinPath(dst, extractFilename(src))):
-                result.code = some(1)
-            else:
+            if canMove:
                 try:
-                    # On OSX, we use POSIX `mv` to bypass restrictions introduced in
-                    # Big Sur on moving files downloaded from the internet
+                    # On OSX, we use POSIX `mv` to bypass restrictions
+                    # introduced in Big Sur on moving files downloaded
+                    # from the internet
                     when defined(macosx):
                         let mvCmd = quoteShellCommand([
                             "mv",
+                            "-f",
                             src, dst
                             ])
                         result.code = some execCmd(mvCmd)
                         if result.code != some 0:
                             raise newException(OSError, "\"" & mvCmd & "\" failed on MacOS ...")
+                        else:
+                            result.code = some(0)
```

Comment:
@gsbabil do you know what this is doing here? I'm assuming it must be something that got mangled in the rebase

Reply:
@bovine3dom I don't remember when this was added, but looks like it can go. Thanks.